test/fuzz/transformation_composite_insert_test.cpp

```diff
@@ -0,0 +1,906 @@
+               OpName %4 "main"
+               OpName %8 "i1"
+               OpName %10 "i2"
+               OpName %12 "base"
+               OpMemberName %12 0 "a1"
+               OpMemberName %12 1 "a2"
+               OpName %14 "b1"
+               OpName %18 "b2"
+               OpName %22 "b3"
+               OpName %26 "lvl1"
+               OpMemberName %26 0 "b1"
+               OpMemberName %26 1 "b2"
+               OpMemberName %26 2 "b3"
+               OpName %28 "l1"
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 1
+          %7 = OpTypePointer Function %6
+          %9 = OpConstant %6 1
+         %11 = OpConstant %6 2
+         %12 = OpTypeStruct %6 %6
+         %13 = OpTypePointer Function %12
+         %26 = OpTypeStruct %12 %12 %12
+         %27 = OpTypePointer Function %26
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+          %8 = OpVariable %7 Function
+         %10 = OpVariable %7 Function
+         %14 = OpVariable %13 Function
+         %18 = OpVariable %13 Function
+         %22 = OpVariable %13 Function
+         %28 = OpVariable %27 Function
+               OpStore %8 %9
+               OpStore %10 %11
+         %15 = OpLoad %6 %8
+         %16 = OpLoad %6 %10
+         %17 = OpCompositeConstruct %12 %15 %16
+               OpStore %14 %17
+         %19 = OpLoad %6 %10
+         %20 = OpLoad %6 %8
+         %21 = OpCompositeConstruct %12 %19 %20
+               OpStore %18 %21
+         %23 = OpLoad %6 %10
+         %24 = OpLoad %6 %10
+         %25 = OpCompositeConstruct %12 %23 %24
+               OpStore %22 %25
+         %29 = OpLoad %12 %14
+         %30 = OpLoad %12 %18
+         %31 = OpLoad %12 %22
+         %32 = OpCompositeConstruct %26 %29 %30 %31
+               OpStore %28 %32
+               OpReturn
+               OpFunctionEnd
+    )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_4;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+  FactManager fact_manager;
+  spvtools::ValidatorOptions validator_options;
+  TransformationContext transformation_context(&fact_manager,
+                                               validator_options);
+  protobufs::InstructionDescriptor instruction_to_insert_before;
+  uint32_t fresh_id, composite_id, object_id;
+  std::vector<uint32_t> index;
+
+  instruction_to_insert_before = MakeInstructionDescriptor(32, SpvOpStore, 0);
+  fresh_id = 50;
+  composite_id = 32;
+  object_id = 9;
+  index = {2, 1};
+  auto transformation_good_1 = TransformationCompositeInsert(
+      instruction_to_insert_before, fresh_id, composite_id, object_id, index);
+  ASSERT_TRUE(transformation_good_1.IsApplicable(context.get(),
+                                                 transformation_context));
+  transformation_good_1.Apply(context.get(), &transformation_context);
+  ASSERT_TRUE(IsValid(env, context.get()));
```

Comment:
Please, add assertions on the synonyms, created by the transformation.

Reply:
I have added them: see line 600.

Reply:
You should then merge this test case with the ones that test synonymous facts. This is to make sure that the fix for the variable not being changed in the loop works.

Reply:
Done, thanks.